diff --git cef/libcef/browser/alloy/alloy_browser_host_impl.cc cef/libcef/browser/alloy/alloy_browser_host_impl.cc
index 4ed837ab6..e8f767afe 100644
--- cef/libcef/browser/alloy/alloy_browser_host_impl.cc
+++ cef/libcef/browser/alloy/alloy_browser_host_impl.cc
@@ -36,6 +36,7 @@
 #include "base/functional/callback_helpers.h"
 #include "chrome/browser/file_select_helper.h"
 #include "chrome/browser/picture_in_picture/picture_in_picture_window_manager.h"
+#include "chrome/browser/ui/exclusive_access/exclusive_access_manager.h"
 #include "components/zoom/page_zoom.h"
 #include "content/browser/gpu/compositor_util.h"
 #include "content/public/browser/desktop_media_id.h"
@@ -1264,6 +1265,49 @@ void AlloyBrowserHostImpl::ResizeDueToAutoResize(content::WebContents* source,
   UpdatePreferredSize(source, new_size);
 }
 
+void AlloyBrowserHostImpl::OnRequestPointerLock(
+    content::WebContents* web_contents,
+    bool user_gesture,
+    bool last_unlocked_by_target,
+    bool allowed) {
+  if (allowed) {
+    exclusive_access_context_impl_->get_manager()->pointer_lock_controller()->RequestToLockPointer(
+        web_contents, user_gesture, last_unlocked_by_target);
+  } else {
+    web_contents->GotResponseToPointerLockRequest(
+        blink::mojom::PointerLockResult::kPermissionDenied);
+  }
+}
+
+void AlloyBrowserHostImpl::RequestPointerLock(content::WebContents* web_contents,
+                                     bool user_gesture,
+                                     bool last_unlocked_by_target) {
+  if (IsWindowless()) {
+    web_contents->GotResponseToPointerLockRequest(
+      blink::mojom::PointerLockResult::kPermissionDenied);
+
+    return;
+  }
+  web_contents->GotResponseToPointerLockRequest(
+    blink::mojom::PointerLockResult::kSuccess);
+}
+
+void AlloyBrowserHostImpl::LostPointerLock() {
+  exclusive_access_context_impl_->get_manager()->pointer_lock_controller()->ExitExclusiveAccessToPreviousState();
+}
+
+void AlloyBrowserHostImpl::RequestKeyboardLock(content::WebContents* web_contents,
+                                      bool esc_key_locked) {
+  exclusive_access_context_impl_->get_manager()->keyboard_lock_controller()->RequestKeyboardLock(
+      web_contents, esc_key_locked);
+}
+
+void AlloyBrowserHostImpl::CancelKeyboardLockRequest(
+    content::WebContents* web_contents) {
+  exclusive_access_context_impl_->get_manager()->keyboard_lock_controller()
+      ->CancelKeyboardLockRequest(web_contents);
+}
+
 void AlloyBrowserHostImpl::RequestMediaAccessPermission(
     content::WebContents* web_contents,
     const content::MediaStreamRequest& request,
@@ -1386,6 +1430,41 @@ void AlloyBrowserHostImpl::WebContentsDestroyed() {
   }
 }
 
+Profile* CfxExclusiveAccessContextImpl::GetProfile() {
+  return nullptr;
+}
+
+bool CfxExclusiveAccessContextImpl::IsFullscreen() const {
+  return base_->contents_delegate()->is_fullscreen();
+}
+
+void CfxExclusiveAccessContextImpl::EnterFullscreen(const GURL& url,
+                                  ExclusiveAccessBubbleType bubble_type,
+                                  const int64_t display_id) {}
+
+void CfxExclusiveAccessContextImpl::ExitFullscreen() {}
+
+void CfxExclusiveAccessContextImpl::UpdateExclusiveAccessExitBubbleContent(
+    const GURL& url,
+    ExclusiveAccessBubbleType bubble_type,
+    ExclusiveAccessBubbleHideCallback bubble_first_hide_callback,
+    bool notify_download,
+    bool force_update) {}
+
+void CfxExclusiveAccessContextImpl::OnExclusiveAccessUserInput() {}
+
+content::WebContents* CfxExclusiveAccessContextImpl::GetActiveWebContents() {
+  return base_->contents_delegate()->web_contents();
+}
+
+bool CfxExclusiveAccessContextImpl::CanUserExitFullscreen() const {
+  return true;
+}
+
+bool CfxExclusiveAccessContextImpl::IsExclusiveAccessBubbleDisplayed() const {
+  return false;
+}
+
 void AlloyBrowserHostImpl::StartAudioCapturer() {
   if (!client_.get() || audio_capturer_) {
     return;
@@ -1409,6 +1488,10 @@ void AlloyBrowserHostImpl::StartAudioCapturer() {
       std::make_unique<CefAudioCapturer>(params, this, audio_handler);
 }
 
+CfxExclusiveAccessContextImpl::CfxExclusiveAccessContextImpl(CefBrowserHostBase* base) : 
+    base_(base), 
+    exclusive_access_manager_(std::make_unique<ExclusiveAccessManager>(this)) {}
+
 // AlloyBrowserHostImpl private methods.
 // -----------------------------------------------------------------------------
 
@@ -1429,7 +1512,8 @@ AlloyBrowserHostImpl::AlloyBrowserHostImpl(
       content::WebContentsObserver(web_contents),
       opener_(kNullWindowHandle),
       is_windowless_(platform_delegate_->IsWindowless()),
-      extension_(extension) {
+      extension_(extension),
+      exclusive_access_context_impl_(std::make_unique<CfxExclusiveAccessContextImpl>(this)) {
   contents_delegate_->ObserveWebContents(web_contents);
 
   if (opener.get() && !is_views_hosted_) {
diff --git cef/libcef/browser/alloy/alloy_browser_host_impl.h cef/libcef/browser/alloy/alloy_browser_host_impl.h
index 2e3c8e78f..e82b6282a 100644
--- cef/libcef/browser/alloy/alloy_browser_host_impl.h
+++ cef/libcef/browser/alloy/alloy_browser_host_impl.h
@@ -22,6 +22,7 @@
 #include "libcef/browser/request_context_impl.h"
 
 #include "base/synchronization/lock.h"
+#include "chrome/browser/ui/exclusive_access/exclusive_access_context.h"
 #include "content/public/browser/web_contents.h"
 #include "content/public/browser/web_contents_delegate.h"
 #include "content/public/browser/web_contents_observer.h"
@@ -30,6 +31,36 @@
 class CefAudioCapturer;
 class CefBrowserInfo;
 class SiteInstance;
+class ExclusiveAccessManager;
+
+class CfxExclusiveAccessContextImpl : public ExclusiveAccessContext {
+ public:
+  CfxExclusiveAccessContextImpl(CefBrowserHostBase* base);
+
+  // ExclusiveAccessContext methods.
+  Profile* GetProfile() override;
+  bool IsFullscreen() const override;
+  void EnterFullscreen(const GURL& url,
+                       ExclusiveAccessBubbleType bubble_type,
+                       const int64_t display_id) override;
+  void ExitFullscreen() override;
+  void UpdateExclusiveAccessExitBubbleContent(
+      const GURL& url,
+      ExclusiveAccessBubbleType bubble_type,
+      ExclusiveAccessBubbleHideCallback bubble_first_hide_callback,
+      bool notify_download,
+      bool force_update) override;
+  void OnExclusiveAccessUserInput() override;
+  content::WebContents* GetActiveWebContents() override;
+  bool CanUserExitFullscreen() const override;
+  bool IsExclusiveAccessBubbleDisplayed() const override;
+
+  ExclusiveAccessManager* get_manager() { return exclusive_access_manager_.get(); }
+
+ private:
+  CefBrowserHostBase* base_;
+  std::unique_ptr<ExclusiveAccessManager> exclusive_access_manager_;
+};
 
 // CefBrowser implementation for the alloy runtime. Method calls are delegated
 // to the CefPlatformDelegate or the WebContents as appropriate. All methods are
@@ -263,6 +294,17 @@ class AlloyBrowserHostImpl : public CefBrowserHostBase,
                            const gfx::Size& pref_size) override;
   void ResizeDueToAutoResize(content::WebContents* source,
                              const gfx::Size& new_size) override;
+  void OnRequestPointerLock(content::WebContents* web_contents,
+                                     bool user_gesture,
+                                     bool last_unlocked_by_target,
+                                     bool allowed);
+  void RequestPointerLock(content::WebContents* web_contents,
+                          bool user_gesture,
+                          bool last_unlocked_by_target) override;
+  void LostPointerLock() override;
+  void RequestKeyboardLock(content::WebContents* web_contents,
+                           bool esc_key_locked) override;
+  void CancelKeyboardLockRequest(content::WebContents* web_contents) override;
   void RequestMediaAccessPermission(
       content::WebContents* web_contents,
       const content::MediaStreamRequest& request,
@@ -352,6 +394,8 @@ class AlloyBrowserHostImpl : public CefBrowserHostBase,
   // starts running when a tab stops being audible, and is canceled if it starts
   // being audible again before it fires.
   std::unique_ptr<base::OneShotTimer> recently_audible_timer_;
+
+  std::unique_ptr<CfxExclusiveAccessContextImpl> exclusive_access_context_impl_;
 };
 
 #endif  // CEF_LIBCEF_BROWSER_ALLOY_ALLOY_BROWSER_HOST_IMPL_H_
