diff --git cef/tools/make_distrib.py cef/tools/make_distrib.py
index 963167a49f9463046dc4ac47517c2401f4b260f8..5ff47cb96b114b43f9a39e06491df72d62c65f31 100644
--- cef/tools/make_distrib.py
+++ cef/tools/make_distrib.py
@@ -334,7 +334,7 @@ def transfer_tools_files(script_dir, build_dirs, output_dir):
         if len(file_name) == 0:
           raise Exception('Failed to parse %s command component: %s' % (mksnapshot_name, cmd))
         if cmd.startswith('../../'):
-          file_path = os.path.realpath(os.path.join(build_dir, cmd))
+          file_path = os.path.abspath(os.path.join(build_dir, cmd))
           # Validate input file/path.
           if not file_path.startswith(src_dir):
             raise Exception('Invalid %s command input file: %s' % (mksnapshot_name, file_path))
@@ -793,10 +793,10 @@ if options.ozone and platform != 'linux':
 script_dir = os.path.dirname(__file__)
 
 # CEF root directory
-cef_dir = os.path.realpath(os.path.join(script_dir, os.pardir))
+cef_dir = os.path.abspath(os.path.join(script_dir, os.pardir))
 
 # src directory
-src_dir = os.path.realpath(os.path.join(cef_dir, os.pardir))
+src_dir = os.path.abspath(os.path.join(cef_dir, os.pardir))
 
 if not git.is_checkout(cef_dir):
   raise Exception('Not a valid checkout: %s' % (cef_dir))
