diff --git third_party/blink/renderer/core/fileapi/file.h third_party/blink/renderer/core/fileapi/file.h
index 72d6b7e109779..e12c4acfd87f1 100644
--- third_party/blink/renderer/core/fileapi/file.h
+++ third_party/blink/renderer/core/fileapi/file.h
@@ -212,6 +212,9 @@ class CORE_EXPORT File final : public Blob {
   }
   const String& name() const { return name_; }
 
+  // Getter for the path IDL attribute.
+  const String& path() const { return GetPath(); }
+
   // Getter for the lastModified IDL attribute,
   // http://dev.w3.org/2006/webapi/FileAPI/#file-attrs
   int64_t lastModified() const;
diff --git third_party/blink/renderer/core/fileapi/file.idl third_party/blink/renderer/core/fileapi/file.idl
index 4339a27e87c12..6aebb1ba4f74a 100644
--- third_party/blink/renderer/core/fileapi/file.idl
+++ third_party/blink/renderer/core/fileapi/file.idl
@@ -31,6 +31,7 @@
 ] interface File : Blob {
     [CallWith=ExecutionContext] constructor(sequence<BlobPart> fileBits, USVString fileName, optional FilePropertyBag options = {});
     readonly attribute DOMString name;
+    readonly attribute DOMString path;
     readonly attribute long long lastModified;
 
     // Non-standard APIs
